frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -251,6 +251,20 @@ function FileManagerMenu:setUpdateItemTable()
             self:exitOrRestart(function() UIManager:restartKOReader() end)
         end,
     }
+    if not Device:isTouchDevice() then
+        --add a shortcut on non touch-device
+        --because this menu is not accessible otherwise
+        self.menu_items.plus_menu = {
+            icon = "resources/icons/appbar.plus.png",
+            remember = false,
+            callback = function()
+                self:onCloseFileManagerMenu()
+                self.ui:tapPlus()
+            end,
+        }
+    else
+        self.menu_items.plus_menu = {}
```

Comment:
Does the top menu not accept it if you just leave it out?

Reply:
It does not.
![2018-03-18-213049_540x720_scrot](https://user-images.githubusercontent.com/3342132/37570733-cc33869e-2af3-11e8-9254-88f85e63ba7b.png)

Took me some time to figure it out. Haven't look in menusorter to try to debug it.

Reply:
Apparently I didn't consider the scenario that a top-level menu item might not be there. :-)

It should suffice to put a simple `if menu_table["KOMenu:menu_buttons"][i].sub_item_table` (or something like that)
https://github.com/koreader/koreader/blob/24c013274430addac6e51b0d9d1c91df0999c132/frontend/ui/menusorter.lua#L134-L136

And of course a unit test, but I'll worry about that — unless you want to. ;-)

Would you mind checking if you can quickly fix the problem as suggested so that we can remove this else branch?

Reply:
Ok, I'm hitting a bit of a road block on this one. I really can't figure out how it all works.

If I add a simple check like you suggested, it works but the next menu are empty. 

I'm trying to dig into the code but I'm progressing very slowly. Maybe we can close this one and I will keep working on it on another PR ?

Reply:
If you want. In its current state it can't be merged yet because it has odd side effects on touch devices. We can also just leave it open and update the content later (whether by adding a commit or by force push).

![screenshot_2018-03-20_21-06-47](https://user-images.githubusercontent.com/202757/37679905-9edd8056-2c82-11e8-9662-80f52e9a5404.png)